app/models/provider.rb

```diff
@@ -62,8 +62,10 @@ class Provider < ApplicationRecord
 
   # TODO: filter to published enrichments, maybe rename to published_address_info
   def address_info
-    (enrichments.with_address_info.last || self)
+    fields = %w[address1 address2 address3 address4 postcode region_code]
+
+    (enrichments.latest_created_at.with_address_info.first || self)
       .attributes_before_type_cast
-      .slice('address1', 'address2', 'address3', 'address4', 'postcode', 'region_code')
+      .slice(*fields)
```

Comment:
Just an opinion, but I think that `fields` is short enough that it can be inline here without getting in the way or readability.